corehq/apps/locations/models.py

```diff
@@ -549,6 +550,43 @@ def _remove_user(self):
             user.active = False
             user.save()
 
+    def deprecate_by(self, other_location, operation, archive=True):
```

Comment:
I think this code should live in `custom.icds` and act on a location, instead of being part of the model.

Reply:
This name seems ambiguous - aren't functions usually verbs? How about `update_deprecation_metadata`, or even just `deprecate`?

Reply:
I am trying as much as possible to not making this a standalone ICDS feature so that it can be used by other projects. Hence you see there is no hard coded location types etc.
But extracting that as a util in `location_rationalization` sounds right, although i like keeping model operations on model itself. It makes it possible to understand the changes a model can have since they are all at one place but we don't really follow that as a pattern.

I usually think of how the method would be called
`location_obj.depreate(other_location, operation)`
`location_obj.deprecated_by(other_location, operation)`
what do you think?

Reply:
Moving to utils so renamed it to `deprecate_location`

Reply:
https://github.com/dimagi/commcare-hq/pull/26898/commits/897e3a6be6ff820317bc5e6d9fa1c2fb0f4707c8